sympy/calculus/tests/test_util.py

```diff
@@ -0,0 +1,35 @@
+from sympy import Symbol, S, oo, sqrt
+from sympy.calculus.util import not_empty_in
+from sympy.sets.sets import Interval, FiniteSet, Complement, Union
+from sympy.abc import x
+
+a = Symbol('a', real=True)
+
+
+def test_not_empty_in():
+    assert not_empty_in(FiniteSet(x, 2*x).intersect(Interval(1, 2,
+                        True, False)), x) == Interval(S(1)/2, 2, True, False)
+    assert not_empty_in(FiniteSet(x, x**2).intersect(Interval(1, 2)), x) == \
+        Union(Interval(-sqrt(2), -1), Interval(1, 2))
+    assert not_empty_in(FiniteSet(x**2 + x, x).intersect(Interval(2, 4)), x) \
+        == Union(Interval(-sqrt(17)/2 - S(1)/2, -2),
+                 Interval(1, -S(1)/2 + sqrt(17)/2), Interval(2, 4))
+    assert not_empty_in(FiniteSet(x/(x - 1)).intersect(S.Reals), x) == \
+        Complement(S.Reals, FiniteSet(1))
+    assert not_empty_in(FiniteSet(a/(a - 1)).intersect(S.Reals), a) == \
+        Complement(S.Reals, FiniteSet(1))
+    assert not_empty_in(FiniteSet((x**2 - 3*x + 2)/(x - 1))
+                        .intersect(S.Reals), x) == \
```

Comment:
I think you can skip the 80 characters limit here, as it looks ugly as of now. Make it in one line.